components/layout_2020/flow/mod.rs

```diff
@@ -1013,19 +1034,35 @@ impl NonReplacedFormattingContext {
                         block_size: block_size.map(|t| t.into()),
                         style: &self.style,
                     },
+                    containing_block,
                 );
-                content_size = LogicalVec2 {
-                    inline: proposed_inline_size,
-                    block: block_size.auto_is(|| {
-                        layout
-                            .content_block_size
-                            .clamp_between_extremums(
-                                min_box_size.block.into(),
-                                max_box_size.block.map(|t| t.into()),
-                            )
-                            .into()
-                    }),
-                };
+
+                if let Some(inline_size) = layout.content_inline_size_for_table {
+                    let outer_inline_size = inline_size + pbm.padding_border_sums.inline;
+                    placement.set_inline_size(outer_inline_size, &pbm);
+                    if outer_inline_size > placement_rect.size.inline {
+                        positioning_context.truncate(&positioning_context_length);
+                        continue;
+                    }
```

Comment:
```suggestion
                    // If this is a table, it's impossible to know the inline size it will take up until after
                    // trying to place it. If the table doesn't fit into this positioning rectangle due to
                    // incompatibility in the inline axis, then retry at another location.
                    if outer_inline_size > placement_rect.size.inline {
                        positioning_context.truncate(&positioning_context_length);
                        continue;
                    }
```

Reply:
I have shortened the lines and added the explanation for always calling `set_inline_size`.